components/webui/imports/ui/SearchView/SearchView.jsx

```diff
@@ -101,13 +110,14 @@ const SearchView = () => {
             ];
         }
 
-        // NOTE: Although we publish and subscribe using the name
-        // `Meteor.settings.public.SearchResultsCollectionName`, the rows are still returned in the
-        // job-specific collection (e.g., "1"); this is because on the server, we're returning a
-        // cursor from the job-specific collection and Meteor creates a collection with the same
-        // name on the client rather than returning the rows in a collection with the published
-        // name.
-        return dbRef.current.getOrCreateCollection(jobId).find({}, findOptions).fetch();
+        resultsCollection.estimatedDocumentCount()
+            .then((count) => {
+                setEstimatedNumResults(
```

Comment:
Update of `estimatedNumResults` is unnecessarily bound with the effects of `[jobId, fieldToSortBy, visibleSearchResultsLimit]`. Let me further optimize this.

Reply:
Addressed by adding a condition check on the job status (`resultsMetadata.lastSignal`).

Note the `useTracker` hook is invoked whenever "there is a change in Dependencies Array", "the component completely re-renders", or **"reactive data source (e.g, collection) changes"**. Therefore we are able to `setEstimatedNumResults` _whenever_ there are updates to the results collection. 
_whenever_ : every `250` ms, since in the `SearchResultsCollection` subscription handler we do `disableOplog` and set `pollingIntervalMs` as `250`.